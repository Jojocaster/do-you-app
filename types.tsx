/**
 * Learn more about using TypeScript with React Navigation:
 * https://reactnavigation.org/docs/typescript/
 */

import { BottomTabScreenProps } from '@react-navigation/bottom-tabs'
import {
  CompositeScreenProps,
  NavigatorScreenParams,
} from '@react-navigation/native'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import { theme } from './src/theme'

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}

export type RootStackParamList = {
  Root: NavigatorScreenParams<RootTabParamList> | undefined
  Modal: undefined
  NotFound: undefined
}

export type RootStackScreenProps<Screen extends keyof RootStackParamList> =
  NativeStackScreenProps<RootStackParamList, Screen>

export type RootTabParamList = {
  TabOne: undefined
  TabTwo: undefined
  More: undefined
}

export type RootTabScreenProps<Screen extends keyof RootTabParamList> =
  CompositeScreenProps<
    BottomTabScreenProps<RootTabParamList, Screen>,
    NativeStackScreenProps<RootStackParamList>
  >

type ThemeInterface = typeof theme

declare module 'styled-components' {
  interface DefaultTheme extends ThemeInterface {}
}
