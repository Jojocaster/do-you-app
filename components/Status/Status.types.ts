export enum StatusType {
  ON,
  OFF,
  LOADING,
}
