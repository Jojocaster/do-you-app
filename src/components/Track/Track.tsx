import { MaterialIcons } from '@expo/vector-icons'
import { format, parseJSON } from 'date-fns'
import { useEffect, useMemo, useState } from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'
import Colors from '../../constants/Colors'
import useColorScheme from '../../hooks/useColorScheme'
import { TrackInfo } from '../../store/slices/tracksInfoSlice'
import { Button } from '../Button/Button'
import * as Clipboard from 'expo-clipboard'
import { Text, View } from '../Themed'

export const Track: React.FC<{
  active: boolean
  onToggle: (track: string) => void
  track: TrackInfo
}> = ({ active, track, onToggle }) => {
  const [copied, setCopied] = useState(false)
  const theme = useColorScheme()
  const timecode = parseJSON(track.played_datetime)
  const formattedTimecode = format(timecode, 'HH:mm')

  const onCopy = () => {
    try {
      Clipboard.setString(`${track.artist} - ${track.title}`)
    } catch (e) {
      console.log('There was an error copying to clipboard.')
    } finally {
      setCopied(true)
    }
  }

  useEffect(() => {
    setCopied(false)
  }, [active])

  // big performance impact, re-rendering all components added a ~200ms delay
  // useMemo prevents that by only re-rendering the tracks involved
  return useMemo(
    () => (
      <View>
        <View style={styles.container} key={track.played_datetime}>
          <View style={styles.timecode}>
            <Text
              style={{
                fontSize: 14,
              }}
            >
              {formattedTimecode}
            </Text>
          </View>
          <TouchableOpacity
            style={{ flex: 1 }}
            onPress={() => onToggle(track.played_datetime)}
          >
            <View style={{ display: 'flex', flexDirection: 'row' }}>
              <View style={{ marginRight: 5 }}>
                <MaterialIcons
                  name={active ? 'expand-less' : 'expand-more'}
                  size={16}
                  color={Colors[theme].text}
                />
              </View>
              <View style={styles.trackContainer}>
                <Text
                  style={{
                    color: Colors[theme].primary,
                    fontWeight: 'bold',
                    fontSize: 14,
                  }}
                >
                  {track.artist}
                </Text>
                <View>
                  <Text>{track.title}</Text>
                </View>

                <View>
                  {active && (
                    <View style={{ marginTop: 10, marginBottom: 10 }}>
                      <Text style={{ fontSize: 12 }}>
                        Confidence:{' '}
                        <Text style={{ fontWeight: 'bold' }}>
                          {track.score / 10}/10
                        </Text>
                      </Text>
                      <Text style={{ fontSize: 12 }}>
                        Album:{' '}
                        <Text style={{ fontWeight: 'bold' }}>
                          {track.album}
                        </Text>
                      </Text>
                      <Text style={{ fontSize: 12 }}>
                        Label:{' '}
                        <Text style={{ fontWeight: 'bold' }}>
                          {track.label}
                        </Text>
                      </Text>
                      <Text style={{ fontSize: 12 }}>
                        Release:{' '}
                        <Text style={{ fontWeight: 'bold' }}>
                          {track.release_date}
                        </Text>
                      </Text>
                      <View
                        style={styles.separator}
                        lightColor="#eee"
                        darkColor="rgba(255,255,255,0.1)"
                      />
                      <View style={{}}>
                        <Button
                          color={copied ? Colors[theme].primary : undefined}
                          icon={copied ? 'check' : 'content-copy'}
                          onPress={onCopy}
                        >
                          {copied ? `Copied` : 'Copy'}
                        </Button>
                      </View>
                    </View>

                    // <View
                    //   style={[
                    //     styles.codeHighlightContainer,
                    //     styles.homeScreenFilename,
                    //   ]}
                    //   darkColor="rgba(255,255,255,0.05)"
                    //   lightColor="rgba(0,0,0,0.05)"
                    // >
                    //   <Text>Album: {track.album}</Text>
                    //   <Text>Release: {track.release_date}</Text>
                    // </View>
                  )}
                </View>
              </View>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    ),
    [active, theme, copied]
  )
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    marginBottom: 10,
  },
  timecode: {
    flexBasis: 60,
  },
  trackContainer: {
    flex: 1,
  },
  codeHighlightContainer: {
    fontSize: 10,
    borderRadius: 3,
    padding: 20,
  },
  homeScreenFilename: {
    marginVertical: 10,
    display: 'flex',
  },
  separator: {
    marginVertical: 10,
    height: 1,
    width: '50%',
  },
})
