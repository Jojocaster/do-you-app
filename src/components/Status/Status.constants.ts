import { StatusType } from './Status.types'

export const StatusLabels: Record<StatusType, string> = {
  [StatusType.OFF]: `We're out.`,
  [StatusType.ON]: `We're on, baby!`,
  [StatusType.LOADING]: `Checking ...`,
  [StatusType.ERROR]: `Checking ...`,
}

export const StatusColours: Record<StatusType, string> = {
  [StatusType.OFF]: `red`,
  [StatusType.ON]: `#27ae60`,
  [StatusType.LOADING]: `orange`,
  [StatusType.ERROR]: `orange`,
}
