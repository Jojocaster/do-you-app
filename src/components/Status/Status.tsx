import { MaterialCommunityIcons } from '@expo/vector-icons'
import React, { useRef, useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { fetchShowInfo, ShowStatus } from '../../store/slices/showSlice'
import { RootState } from '../../store/store'
import { Text, View } from '../Themed'
import { StatusColours, StatusLabels } from './Status.constants'

export const Status: React.FC = () => {
  const dispatch = useDispatch()
  const { status, lastUpdated } = useSelector((state: RootState) => state.show)
  const timeout = useRef<NodeJS.Timeout>()

  const checkStatus = async () => {
    if (!lastUpdated) {
      dispatch(fetchShowInfo())
    } else {
      const now = new Date().getTime()
      const diff = now - lastUpdated
      const diffInMinutes = diff / (1000 * 60)
      if (diffInMinutes > 1) {
        dispatch(fetchShowInfo())
      }
    }
  }

  useEffect(() => {
    checkStatus()
  }, [])

  // check status every minute if not in loading state
  //TODO: turn into background service, check against schedule to avoid hammering the API
  useEffect(() => {
    if (status !== ShowStatus.LOADING) {
      // reset timeout
      if (timeout.current) {
        clearTimeout(timeout.current)
      }

      timeout.current = setTimeout(() => {
        checkStatus()
      }, 1000 * 60)
    }
  }, [status])

  return (
    <View
      style={{
        paddingTop: 50,
        paddingBottom: 20,
        justifyContent: 'space-between',
        alignItems: 'center',
        backgroundColor: '#121212',
        display: 'flex',
        flexDirection: 'row',
        paddingHorizontal: 20,
      }}
    >
      <Text style={{ fontWeight: 'bold' }}>Status: </Text>
      <View
        style={{
          backgroundColor: 'transparent',
          display: 'flex',
          flexDirection: 'row',

          alignItems: 'center',
        }}
      >
        <Text>{StatusLabels[status]}</Text>
        <MaterialCommunityIcons
          name="adjust"
          size={20}
          style={{ marginLeft: 5 }}
          color={StatusColours[status]}
        />
      </View>
    </View>
  )
}
