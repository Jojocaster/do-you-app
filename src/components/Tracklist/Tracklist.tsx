import React, { useState } from 'react'
import { FlatList } from 'react-native'
import { TrackInfo } from '../../store/slices/tracksInfoSlice'
import { Track } from '../Track/Track'

export const Tracklist: React.FC<{
  tracks: TrackInfo[]
  showStart?: string
  virtual?: boolean
}> = ({ showStart, tracks, virtual = true }) => {
  const [activeTrack, setActiveTrack] = useState<string>()

  const onToggle = (track: string) => {
    if (track === activeTrack) {
      setActiveTrack(undefined)
    } else {
      setActiveTrack(track)
    }
  }

  if (!virtual) {
    return (
      <>
        {tracks.map((t, i) => (
          <Track
            key={i}
            showStart={showStart}
            active={activeTrack === t.played_datetime}
            onToggle={onToggle}
            track={t}
          />
        ))}
      </>
    )
  }

  return (
    <FlatList
      data={tracks}
      renderItem={(t) => (
        <Track
          showStart={showStart}
          active={activeTrack === t.item.played_datetime}
          onToggle={onToggle}
          track={t.item}
        />
      )}
      keyExtractor={(track) => track.played_datetime}
      showsVerticalScrollIndicator={false}
      fadingEdgeLength={100}
      overScrollMode="never"
      style={{ width: '100%' }}
    />
  )
}
