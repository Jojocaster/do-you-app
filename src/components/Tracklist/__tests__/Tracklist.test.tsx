import { fireEvent, waitFor } from '@testing-library/react-native'
import { FlatList } from 'react-native'
import renderer, { act } from 'react-test-renderer'
import { StatusType } from '../../Status/Status.types'
import { Track } from '../../Track/Track'
import { Tracklist } from '../Tracklist'

const mockStore = {
  settings: {
    darkTheme: false,
  },
  tracksInfo: { loading: true, tracks: [] },
  show: {
    status: StatusType.OFF,
  },
}

// mock for expo icons
jest.mock(
  '@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js',
  () => {
    return () => ''
  }
)
jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn((cb) => cb(mockStore)),
}))

jest.mock('../../../hooks/useColorScheme', () => jest.fn(() => 'dark'))

const mockTracks = [
  {
    artist: 'Hipnotic',
    played_date: '2022-07-27',
    timecode: '00:03',
    song_link: 'https://lis.tn/uoxrjY',
    label: 'Essential Media Group',
    score: 64,
    received_datetime: '2022-07-27 09:23:13+0100',
    album: 'The Soul Of Philadelphia',
    release_date: '2008-09-09',
    played_datetime: '2022-07-27 09:23:04+0100',
    out_of: 1,
    title: 'Are You Lonely?',
  },
  {
    artist: 'Larry Wu',
    played_date: '2022-07-27',
    timecode: '00:15',
    song_link: 'https://lis.tn/Zgqvv',
    label: 'SME - Sony Music Entertainment',
    score: 100,
    received_datetime: '2022-07-27 09:33:51+0100',
    album: 'Grand 12 inches 13',
    release_date: '2015-05-29',
    played_datetime: '2022-07-27 09:33:41+0100',
    out_of: 1,
    title: 'Let Me Show You (Long Vocal Version)',
  },
  {
    artist: 'John Kongos',
    played_date: '2022-07-27',
    timecode: '00:08',
    song_link: 'https://lis.tn/XFHWU',
    label: 'MERLIN - Fabric Records',
    score: 100,
    received_datetime: '2022-07-27 09:45:35+0100',
    album: 'fabric presents Danilo Plessow (MCDE)',
    release_date: '2021-05-28',
    played_datetime: '2022-07-27 09:45:25+0100',
    out_of: 2,
    title: "I'm Dreaming (Any Moment I May Wake Up Screaming)",
  },
  {
    artist: 'Dreams Unlimited',
    played_date: '2022-07-27',
    timecode: '05:39',
    song_link: 'https://lis.tn/DEEPINYOULTJCLUBMIX',
    label: 'Broma16',
    score: 100,
    received_datetime: '2022-07-27 09:51:52+0100',
    album: 'The Best of DREAMS UNLIMITED',
    release_date: '2019-04-01',
    played_datetime: '2022-07-27 09:51:43+0100',
    out_of: 1,
    title: 'DEEP IN YOU (LTJ CLUB MIX)',
  },
]

describe('Tracklist', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(<Tracklist tracks={mockTracks} />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should render on `Track` component per item', () => {
    const wrapper = renderer.create(<Tracklist tracks={mockTracks} />)
    expect(wrapper.root.findAllByType(Track).length).toEqual(mockTracks.length)
  })

  describe('when `virtual` is set to false', () => {
    it('should NOT use a Flatlist', () => {
      const wrapper = renderer.create(
        <Tracklist virtual={false} tracks={mockTracks} />
      )
      const flatlist = wrapper.root.findAllByType(FlatList)
      expect(flatlist.length).toBe(0)
    })
  })

  describe('when `virtual` is set to true', () => {
    it('should use a Flatlist', () => {
      const wrapper = renderer.create(
        <Tracklist virtual={true} tracks={mockTracks} />
      )
      const flatlist = wrapper.root.findAllByType(FlatList)
      expect(flatlist.length).toBe(1)
    })
  })

  // describe('when a track is pressed', () => {
  //   it('should set its `active` prop to true', async () => {
  //     const wrapper = renderer.create(<Tracklist tracks={mockTracks} />)
  //     const item = wrapper.root.findAllByType(Track)[0]
  //     expect(item.props.active).toBeFalsy()

  //     act(() => {
  //       item.props.onToggle(mockTracks[0])
  //     })
  //     await waitFor(() => expect(item.props.active).toBeTruthy())
  //   })
  // })
})
