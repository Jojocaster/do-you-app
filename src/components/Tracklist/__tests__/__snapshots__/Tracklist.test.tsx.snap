// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracklist should render correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "album": "The Soul Of Philadelphia",
        "artist": "Hipnotic",
        "label": "Essential Media Group",
        "out_of": 1,
        "played_date": "2022-07-27",
        "played_datetime": "2022-07-27 09:23:04+0100",
        "received_datetime": "2022-07-27 09:23:13+0100",
        "release_date": "2008-09-09",
        "score": 64,
        "song_link": "https://lis.tn/uoxrjY",
        "timecode": "00:03",
        "title": "Are You Lonely?",
      },
      Object {
        "album": "Grand 12 inches 13",
        "artist": "Larry Wu",
        "label": "SME - Sony Music Entertainment",
        "out_of": 1,
        "played_date": "2022-07-27",
        "played_datetime": "2022-07-27 09:33:41+0100",
        "received_datetime": "2022-07-27 09:33:51+0100",
        "release_date": "2015-05-29",
        "score": 100,
        "song_link": "https://lis.tn/Zgqvv",
        "timecode": "00:15",
        "title": "Let Me Show You (Long Vocal Version)",
      },
      Object {
        "album": "fabric presents Danilo Plessow (MCDE)",
        "artist": "John Kongos",
        "label": "MERLIN - Fabric Records",
        "out_of": 2,
        "played_date": "2022-07-27",
        "played_datetime": "2022-07-27 09:45:25+0100",
        "received_datetime": "2022-07-27 09:45:35+0100",
        "release_date": "2021-05-28",
        "score": 100,
        "song_link": "https://lis.tn/XFHWU",
        "timecode": "00:08",
        "title": "I'm Dreaming (Any Moment I May Wake Up Screaming)",
      },
      Object {
        "album": "The Best of DREAMS UNLIMITED",
        "artist": "Dreams Unlimited",
        "label": "Broma16",
        "out_of": 1,
        "played_date": "2022-07-27",
        "played_datetime": "2022-07-27 09:51:43+0100",
        "received_datetime": "2022-07-27 09:51:52+0100",
        "release_date": "2019-04-01",
        "score": 100,
        "song_link": "https://lis.tn/DEEPINYOULTJCLUBMIX",
        "timecode": "05:39",
        "title": "DEEP IN YOU (LTJ CLUB MIX)",
      },
    ]
  }
  disableVirtualization={false}
  fadingEdgeLength={100}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  overScrollMode="never"
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "width": "100%",
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1c1c1c",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#1c1c1c",
              },
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#1c1c1c",
                },
                Object {
                  "flexBasis": 60,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontSize": 14,
                  },
                ]
              }
            >
              10:23
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#1c1c1c",
                  },
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1c1c1c",
                    },
                    Object {
                      "marginRight": 5,
                    },
                  ]
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1c1c1c",
                    },
                    Object {
                      "flex": 1,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                      },
                      Object {
                        "color": "#FFDC3B",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Hipnotic
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1c1c1c",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                        },
                        undefined,
                      ]
                    }
                  >
                    Are You Lonely?
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1c1c1c",
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1c1c1c",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#1c1c1c",
              },
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#1c1c1c",
                },
                Object {
                  "flexBasis": 60,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontSize": 14,
                  },
                ]
              }
            >
              10:33
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#1c1c1c",
                  },
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1c1c1c",
                    },
                    Object {
                      "marginRight": 5,
                    },
                  ]
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1c1c1c",
                    },
                    Object {
                      "flex": 1,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                      },
                      Object {
                        "color": "#FFDC3B",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Larry Wu
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1c1c1c",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                        },
                        undefined,
                      ]
                    }
                  >
                    Let Me Show You (Long Vocal Version)
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1c1c1c",
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1c1c1c",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#1c1c1c",
              },
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#1c1c1c",
                },
                Object {
                  "flexBasis": 60,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontSize": 14,
                  },
                ]
              }
            >
              10:45
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#1c1c1c",
                  },
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1c1c1c",
                    },
                    Object {
                      "marginRight": 5,
                    },
                  ]
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1c1c1c",
                    },
                    Object {
                      "flex": 1,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                      },
                      Object {
                        "color": "#FFDC3B",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  John Kongos
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1c1c1c",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                        },
                        undefined,
                      ]
                    }
                  >
                    I'm Dreaming (Any Moment I May Wake Up Screaming)
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1c1c1c",
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1c1c1c",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#1c1c1c",
              },
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#1c1c1c",
                },
                Object {
                  "flexBasis": 60,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontSize": 14,
                  },
                ]
              }
            >
              10:51
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#1c1c1c",
                  },
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1c1c1c",
                    },
                    Object {
                      "marginRight": 5,
                    },
                  ]
                }
              >
                <Text />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#1c1c1c",
                    },
                    Object {
                      "flex": 1,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                      },
                      Object {
                        "color": "#FFDC3B",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Dreams Unlimited
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1c1c1c",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#fff",
                        },
                        undefined,
                      ]
                    }
                  >
                    DEEP IN YOU (LTJ CLUB MIX)
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#1c1c1c",
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
